### DISCUSSION TITLE
Clean x-model on removing inputs with x-if

### DISCUSSION THREAD

**Discussion by and-cesbo** [alpinejs/alpine#3113] [1. Help]
Hi,

Example: https://codepen.io/cesbo/pen/eYMXgGj

I have a dynamic form with elements depend on others.
When checkbox is turned on, form has two additional inputs in `<template x-if="..."><div>...</div></template>`.
Is there some way to clean x-model values binded to this additional inputs when they removed?

I found two methods:
1. `@change` handler that remove all unneded variables from model. This method looks bad on large forms with many fields
2. Good old form serialization on `@submit`. Iterate all available elements and store data to new object. This method is good but needed additional code to

But maybe some other solution.
Thanks!

**Comment by and-cesbo** [comment]
I found solution. It looks good for me:

```
Alpine.directive('cleanup', (el, {}, { cleanup }) => {
	cleanup(_ => {
		el._x_model.set(undefined)
	})
})
```

Usage: `<input type="text" x-model="data.key" x-cleanup />`

so, when element removed from the DOM, then related model cleaned.
maybe it will be useful as modifier for `x-model` i can prepare pull request for that.

**Reply by Tim-Wils** [reply]
Nice solution!

**Reply by ekwoka** [reply]
This looks like instead it should be implemented as the basic behavior of x-model to self-cleanup.
